frontend/src/App.tsx

```diff
@@ -915,12 +918,13 @@ export class App extends PureComponent<Props, State> {
       "streamlit-wide": userSettings.wideMode,
     })
 
-    const renderedDialog: React.ReactNode = dialog
-      ? StreamlitDialog({
-          ...dialog,
-          onClose: this.closeDialog,
-        })
-      : null
+    const renderedDialog: React.ReactNode =
+      dialog && !this.props.s4aCommunication.currentState.forcedModalClose
```

Comment:
Hm, will this prevent a user from opening a dialog after they get the first `CLOSE_MODAL` message from s4a?

I'm not sure if a `StreamlitDialog` that gets unrendered because `this.props.s4aCommunication.currentState.forcedModalClose === true` will end up calling its `onClose` handler, which I think is the only way that `forcedModalClose` ends up getting set back to `false`.